### DISCUSSION TITLE
Debouncing effect suppression with a glitch filter

### DISCUSSION THREAD

**Discussion by mnmneE** [wehr-lab/autopilot#125] [Q&A] [answered]
Hi there,

I need to implement a simple task with 3 push buttons, and for that, I must use callback functions,
yet the functions are called repeatedly because of debouncing effect,
I didn't find in the documentation of a Digital_In object a method that may help, but depending on pigpio library it is possible to use a glitch filter with the associated pin; the problem is that when I  tried to set that filter using the following command:

''self.hardware['POKES'].['PB1'].set_glitch_filter(23, 100)''
the pilot terminal throws an attribute error;
is there a way for setting up this kind of filters or any other useful method to deal with debouncing effect;

in advance thanks very much for your help.

Regards

( as an aside, I didn't include the callback functions in the iteration cycle )

**Comment by mnmneE** [comment, marked as answer]
just by rereading the documents of GPIO  class , I think the filter will work with adding PIG connection; so that the instruction should be;
''self.hardware['POKES'].['PB1'].pig.set_glitch_filter(23, 100)''

**Comment by sneakers-the-rat** [comment]
makes me think GPIO should override `__getattr__`  to check if the pigpio client has a corresponding method

something like

```python
def __getattr__(self, attr):
    try:
        return super(self).__getattr__(self, attr)
    except AttributeError as e:
        try:
            return getattr(self.pig, attr)
        except AttributeError:
            raise e
       
```
